octoprint_slic3r/profile.py

```diff
@@ -239,11 +261,18 @@ def to_slic3r_ini(cls, profile, path, display_name=None, description=None):
 	@classmethod
 	def convert_value(cls, key, value, default, sep=","):
 		try:
-			if key in float_or_percentage:
+			'''Some defaults can be instances of int whereas its values are instances of float. 
```

Comment:
What I mean here is that you will only run into a problem if `default` is of type `int` but the `value` is a string representing a float. The question is why the `default` is an `int` in the first place if the `value` may very well be a float.

So what I'd suggest here is rather have the default of keys that can possibly be float also be float instead of int. Example: `small_perimeter_speed = 30.0` instead of `small_perimeter_speed = 30`. That way `default` will be of type `float` already and all is well.

But I might not be understanding the problem correctly and there might be values which can be float and int in malformed profiles that Slic3r will still happily accept, in which case just ignore this :)